app/RouteHandlers/ShowDonationConfirmationHandler.php

```diff
@@ -36,12 +39,42 @@ public function handle( ParameterBag $params ): Response {
 				$this->ffFactory->newDonationConfirmationPresenter()->present(
 					$responseModel->getDonation(),
 					$params->get( 'updateToken', '' ),
-					$selectedConfirmationPage
+					$selectedConfirmationPage,
+					$this->collectPiwikTrackingEvents( $sessionTrackingData, $responseModel->getDonation() )
 				)
 			);
 		}
 
 		throw new AccessDeniedException( 'access_denied_donation_confirmation' );
 	}
 
+	private function collectPiwikTrackingEvents( array $sessionTrackingData, Donation $donation ) {
+		$piwikTracking = new PiwikCustomTracking();
+		if ( isset( $sessionTrackingData['paymentType'] ) ) {
+			$piwikTracking->setCustomVariable(
+				1,
+				'Payment',
+				$sessionTrackingData['paymentType'] . '/' . $donation->getPaymentType(),
+				'visit'
+			);
+		}
+		if ( isset( $sessionTrackingData['paymentAmount'] ) ) {
+			$piwikTracking->setCustomVariable(
+				2,
+				'Amount',
+				$sessionTrackingData['paymentAmount'] . '/' . $donation->getAmount()->getEuroString(),
+				'visit'
+			);
+		}
+		if ( isset( $sessionTrackingData['paymentInterval'] ) ) {
+			$piwikTracking->setCustomVariable(
+				3,
+				'Interval',
+				$sessionTrackingData['paymentInterval'] . '/' . $donation->getPaymentIntervalInMonths(),
+				'visit'
+			);
+		}
+		return $piwikTracking;
+	}
+
```

Comment:
Mmmh. Actually the `AddDonationUseCase` should forward to the `ShowDonationConfirmationUseCase`. If we do that, collecting the custom variables is only needed there (unless we do more tracking in other use cases).